### PR TITLE
chore(transport): Use hyper_util::service::TowerToHyperService

### PR DESCRIPTION
**Pull request by tottoto**
Uses `hyper_util::service::TowerToHyperService` instead of defining the own version with change.

### CODE REVIEW

File: tonic/src/transport/server/mod.rs [hunk 1]
```diff
@@ -595,7 +598,9 @@ impl<L> Server<L> {
                     let req_svc = svc
                         .call(&io)
                         .await
-                        .map_err(super::Error::from_source)?;
+                        .map_err(super::Error::from_source)?
```
Comment (djc) [inline tonic/src/transport/server/mod.rs:601]: Style nit: how do you feel about using an explicit `match` here? I generally feel like `map().map_err()` doesn't give as much clarity.

**Review by djc — APPROVED** [review]
Nice, thanks!

File: tonic/src/transport/server/mod.rs (hunk 1 above)
Comment (tottoto) [reply tonic/src/transport/server/mod.rs:601]: Personally, I feel the style using method chain like this to be a bit more intuitive.

Comment (tottoto) [reply tonic/src/transport/server/mod.rs:601]: I feel the style using method chain like this to be a bit more intuitive.

**Pull request merged**